src/components/synapse.rs

```diff
@@ -151,6 +157,40 @@ impl SynapseComponent {
         .await
     }
 
+    #[tracing::instrument(name = "put send message event to the given room > Synapse components")]
+    pub async fn send_message_event_given_room(
+        &self,
+        token: &str,
+        room_id: &str,
+        room_event: FriendshipEvent,
+        room_message_body: &str,
+    ) -> Result<RoomEventResponse, CommonError> {
+        // The transaction ID for this event.
+        // Clients should generate an ID unique across requests with the same access token;
+        // it will be used by the server to ensure idempotency of requests.
+        // https://github.com/matrix-org/matrix-js-sdk/blob/1842004db2c9a58896fe159d52d489cf6ee3fcf7/src/client.ts#L3818
+        let txn_id = format!(
```

Comment:
I copied this from [matrix-js-sdk](https://github.com/matrix-org/matrix-js-sdk/blob/1842004db2c9a58896fe159d52d489cf6ee3fcf7/src/client.ts#L3818). Not sure if it's the correct thing to do.